packages/cli/src/controllers/dynamicNodeParameters.controller.ts

```diff
@@ -1,54 +1,29 @@
-import type { RequestHandler } from 'express';
-import { NextFunction, Response } from 'express';
-import type {
-	INodeListSearchResult,
-	INodePropertyOptions,
-	ResourceMapperFields,
-} from 'n8n-workflow';
+import type { INodePropertyOptions } from 'n8n-workflow';
 import { jsonParse } from 'n8n-workflow';
 
-import { Get, Middleware, RestController } from '@/decorators';
+import { Post, RestController } from '@/decorators';
 import { getBase } from '@/WorkflowExecuteAdditionalData';
 import { DynamicNodeParametersService } from '@/services/dynamicNodeParameters.service';
 import { DynamicNodeParametersRequest } from '@/requests';
 import { BadRequestError } from '@/errors/response-errors/bad-request.error';
 
-const assertMethodName: RequestHandler = (req, res, next) => {
-	const { methodName } = req.query as DynamicNodeParametersRequest.BaseRequest['query'];
-	if (!methodName) {
-		throw new BadRequestError('Parameter methodName is required.');
-	}
-	next();
-};
-
 @RestController('/dynamic-node-parameters')
 export class DynamicNodeParametersController {
 	constructor(private readonly service: DynamicNodeParametersService) {}
 
-	@Middleware()
-	parseQueryParams(req: DynamicNodeParametersRequest.BaseRequest, _: Response, next: NextFunction) {
-		const { credentials, currentNodeParameters, nodeTypeAndVersion } = req.query;
-		if (!nodeTypeAndVersion) {
-			throw new BadRequestError('Parameter nodeTypeAndVersion is required.');
-		}
-		if (!currentNodeParameters) {
-			throw new BadRequestError('Parameter currentNodeParameters is required.');
-		}
-
-		req.params = {
-			nodeTypeAndVersion: jsonParse(nodeTypeAndVersion),
-			currentNodeParameters: jsonParse(currentNodeParameters),
-			credentials: credentials ? jsonParse(credentials) : undefined,
-		};
+	@Post('/options')
+	async getOptions(req: DynamicNodeParametersRequest.Options): Promise<INodePropertyOptions[]> {
+		const {
+			credentials,
+			currentNodeParameters,
+			nodeTypeAndVersion,
+			path,
+			methodName,
+			loadOptions,
+		} = req.body;
 
-		next();
-	}
+		if (!methodName) throw new BadRequestError('Missing `methodName` in request body');
 
-	/** Returns parameter values which normally get loaded from an external API or get generated dynamically */
-	@Get('/options')
-	async getOptions(req: DynamicNodeParametersRequest.Options): Promise<INodePropertyOptions[]> {
-		const { path, methodName, loadOptions } = req.query;
-		const { credentials, currentNodeParameters, nodeTypeAndVersion } = req.params;
 		const additionalData = await getBase(req.user.id, currentNodeParameters);
 
 		if (methodName) {
```

Comment:
is this check still needed now that we are also checking `methodName` on line 25?

Reply:
Good catch, please see [e6d3049](https://github.com/n8n-io/n8n/pull/9384/commits/e6d3049ae94500b55600f77597f1ca6978dd615b)